zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ReplayStateMachine.java

```diff
@@ -210,7 +210,7 @@ private boolean tryToReplayBatch(final Batch batch) throws Exception {
           batch.forEachRemaining(this::replayEvent);
 
           if (batchSourceEventPosition > snapshotPosition) {
-            lastProcessedPositionState.markAsProcessed(batchSourceEventPosition);
+            lastProcessedPositionState.markAsProcessed(lastReadRecordPosition);
```

Comment:
🔧 I don't think this is supposed to be in here.

Reply:
Oh wow, that should not have slipped in here :grimacing: